database/src/main/java/com/pip/chatbot/repository/CitiesRepository.java

```diff
@@ -14,43 +14,41 @@ public CitiesRepository(DSLContext dsl) {
         this.dsl = dsl;
     }
 
-    public City getCity(String cityName) throws IllegalArgumentException {
-        City city = dsl.selectFrom(Tables.CITY).where(Tables.CITY.CITY_.eq(cityName)).fetchAnyInto(City.class);
-        if (city == null) {
-            throw new IllegalArgumentException();
-        }
-        return city;
+    public Optional<City> getCity(String cityName) {
+        return Optional.ofNullable(dsl.selectFrom(Tables.CITY)
+                .where(Tables.CITY.CITY_.eq(cityName))
+                .fetchAnyInto(City.class));
+    }
+
+    public boolean isCityExist(String city) {
+        return dsl.fetchExists(dsl.selectFrom(Tables.CITY).where(Tables.CITY.CITY_.eq(city)));
     }
 
     public List<City> getAllCities() {
         return dsl.selectFrom(Tables.CITY).fetchInto(City.class);
     }
 
-    public City createCity(City city) {
-        CityRecord cityRecord = dsl.newRecord(Tables.CITY);
-
-        cityRecord.setCity(city.getCity());
-        cityRecord.setCountry(city.getCountry());
-        cityRecord.setLatitude(city.getLatitude());
-        cityRecord.setLongitude(city.getLongitude());
-        cityRecord.store();
-        System.out.println(cityRecord.into(City.class));
-        return cityRecord.into(City.class);
+    public Optional<City> createCity(City city) {
+        return Optional.ofNullable(dsl.insertInto(Tables.CITY)
```

Comment:
Tutaj akurat optionala nie zwracajmy. Jak nie wyjdzie create to i tak pójdzie exception:)